src/components/diagrams/diagram.js

```diff
@@ -0,0 +1,911 @@
+import AddCircleIcon from '@mui/icons-material/AddCircle';
+import RemoveCircleIcon from '@mui/icons-material/RemoveCircle';
+
+import { AutoSizer } from 'react-virtualized';
+
+import { useIntl } from 'react-intl';
+
+import clsx from 'clsx';
+import { RunningStatus } from '../util/running-status';
+import AlertInvalidNode from '../util/alert-invalid-node';
+import MinimizeIcon from '@mui/icons-material/Minimize';
+import PushPinIcon from '@mui/icons-material/PushPin';
+import PushPinOutlinedIcon from '@mui/icons-material/PushPinOutlined';
+import BaseEquipmentMenu from '../menus/base-equipment-menu';
+import withEquipmentMenu from '../menus/equipment-menu';
+import withLineMenu from '../menus/line-menu';
+import { equipments } from '../network/network-equipments';
+import { useIntlRef, useSnackMessage } from '@gridsuite/commons-ui';
+import { useIsAnyNodeBuilding } from '../util/is-any-node-building-hook';
+import Alert from '@mui/material/Alert';
+import { isNodeBuilt, isNodeReadOnly } from '../graph/util/model-functions';
+import {
+    NetworkAreaDiagramViewer,
+    SingleLineDiagramViewer,
+} from '@powsybl/diagram-viewer';
+import {
+    SvgType,
+    getEquipmentTypeFromFeederType,
+    useDiagram,
+    computePaperAndSvgSizesIfReady,
+    commonDiagramStyle,
+    commonSldStyle,
+    commonNadStyle,
+    MAX_HEIGHT_SUBSTATION,
+    MAX_HEIGHT_VOLTAGE_LEVEL,
+    MAX_HEIGHT_NETWORK_AREA_DIAGRAM,
+    MAX_WIDTH_SUBSTATION,
+    MAX_WIDTH_VOLTAGE_LEVEL,
+    MAX_WIDTH_NETWORK_AREA_DIAGRAM,
+    NoSvg,
+    LOADING_WIDTH,
+} from './diagram-common';
+import makeStyles from '@mui/styles/makeStyles';
+
+const customSldStyle = (theme) => {
+    return {
+        '& .arrow': {
+            fill: theme.palette.text.primary,
+        },
+    };
+};
+
+const useStyles = makeStyles((theme) => ({
+    divSld: commonSldStyle(theme, customSldStyle(theme)),
+    divNad: commonNadStyle(theme),
+    ...commonDiagramStyle(theme),
+}));
+
+let initialWidth, initialHeight;
+
+const Diagram = forwardRef((props, ref) => {
+    const [svg, setSvg] = useState(NoSvg);
+    const svgUrl = useRef(''); // TODO CHARLY dans Nad, utilisé comme Props
+    const svgDraw = useRef();
+    const dispatch = useDispatch();
+    const { snackError } = useSnackMessage();
+    const intlRef = useIntlRef();
+    const svgRef = useRef();
+    const theme = useTheme();
+    const classes = useStyles();
+    const intl = useIntl();
+    const {
+        computedHeight,
+        //currentNode, // on utilise plutôt celui du SLD : via le useSelector
+        depth,
+        disabled,
+        loadFlowStatus,
+        numberToDisplay,
+        onMinimize,
+        onTogglePin,
+        pinned,
+        setDisplayedDiagramHeights,
+        diagramId,
+        svgType,
+        //onClose,
+        setDepth,
+        totalHeight,
+        totalWidth,
+    } = props;
+
+    const [, , , closeDiagramView] = useDiagram(); // We only use this function (the fourth)
```

Comment:
you can replace by `const { closeDiagramView } = useDiagram();`

Reply:
Yes, and I also changed the return value of useDiagram.